pkg/option/config.go

```diff
@@ -3539,6 +3548,17 @@ func (c *DaemonConfig) Populate(vp *viper.Viper) {
 	c.PolicyCIDRMatchMode = vp.GetStringSlice(PolicyCIDRMatchMode)
 	c.EnableNodeSelectorLabels = vp.GetBool(EnableNodeSelectorLabels)
 	c.NodeLabels = vp.GetStringSlice(NodeLabels)
+
+	// Parse node label patterns
+	nodeLabelPatterns := vp.GetStringSlice(ExcludeNodeLabelPatterns)
+	for _, pattern := range nodeLabelPatterns {
+		r, err := regexp.Compile(pattern)
+		if err != nil {
+			log.WithError(err).Errorf("Unable to compile exclude node label regex pattern %s", pattern)
+			continue
```

Comment:
I'm wondering whether we should bail out hard in case the regex pattern cannot be compiled, to make sure that users can easily catch broken configurations.

Reply:
Hmm. I feel like an error level log should provide enough signal to cluster operators ? Since this flag is effectively only an optimization to avoid unnecessary writes, this should be enough ?